plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaTable.java

```diff
@@ -0,0 +1,213 @@
+
+import com.google.common.collect.ImmutableList;
+import io.trino.plugin.deltalake.transactionlog.MetadataEntry;
+import io.trino.plugin.deltalake.transactionlog.ProtocolEntry;
+import jakarta.annotation.Nullable;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.MoreCollectors.onlyElement;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getCheckConstraints;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnComments;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnInvariants;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnTypes;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnsMetadata;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnsNullability;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getExactColumnNames;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getGeneratedColumnExpressions;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaTable
+{
+    private final List<Column> columns;
+    private final List<String> constraints;
+
+    private DeltaTable(List<Column> columns, List<String> constraints)
+    {
+        requireNonNull(columns, "columns is null");
+        requireNonNull(constraints, "columns is null");
+        checkArgument(!columns.isEmpty(), "columns must not be empty");
+
+        this.columns = new ArrayList<>(columns);
+        this.constraints = ImmutableList.copyOf(constraints);
+    }
+
+    public List<String> columnNames()
+    {
+        return columns.stream()
+                .map(column -> column.name)
+                .collect(toImmutableList());
+    }
+
+    @NotNull
+    public Object getColumnType(String name)
+    {
+        return findColumn(name).type;
+    }
+
+    public boolean isNullable(String name)
+    {
+        return findColumn(name).nullable;
+    }
+
+    @Nullable
+    public String getColumnComment(String name)
+    {
+        return findColumn(name).comment.orElse(null);
+    }
+
+    @Nullable
+    public Map<String, Object> getColumnMetadata(String name)
+    {
+        return findColumn(name).metadata;
+    }
+
+    public Optional<String> getColumnGeneration(String name)
+    {
+        return findColumn(name).generation;
+    }
+
+    public List<String> constraints()
+    {
+        return constraints;
+    }
+
+    private Column findColumn(String name)
```

Comment:
why is this method `private` ?

in `io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport#serializeStructType(io.trino.plugin.deltalake.DeltaTable)` we could find the "column" first and use it instead of calling "deltaTable.getColumnXXX(columnName)` methods.

Reply:
Answered in https://github.com/trinodb/trino/pull/18133#discussion_r1493050064